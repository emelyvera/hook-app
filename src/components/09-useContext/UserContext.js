import { createContext } from 'react';

// este es un hired order component, eso es que tene coimponentes hijos 
export const UserContext = createContext( null );