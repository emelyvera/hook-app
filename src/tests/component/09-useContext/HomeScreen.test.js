import React from "react";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { mount } from 'enzyme';
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Pruebas en el componente <HomeScreen />', () => {
  
    const user ={
        name: 'emely',
        email: 'emel@jbfdg'
    };

    const wrapper =mount (
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen/>
        </UserContext.Provider>
    )

    test('should de mostrarse correctamente', () => {
      
        expect(wrapper).toMatchSnapshot();

    });
    
});
