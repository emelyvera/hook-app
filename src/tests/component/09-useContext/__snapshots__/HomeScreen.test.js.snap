// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <HomeScreen /> should de mostrarse correctamente 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
    "name": "emely",
    "email": "emel@jbfdg"
}
  </pre>
</div>
`;
